Upgrade hiredis

jessicamack opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
The requirement was previously blocked because the package didn't provide source distribution on PyPi but this has been resolved https://pypi.org/project/hiredis/2.3.2/#files
<!---
If you are fixing an existing issue, please include "related #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
 - Other

##### AWX VERSION
<!--- Paste verbatim output from `make VERSION` between quotes below -->
```
awx: 24.6.2
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

jessicamack committed:
upgrade hiredis

was previously blocked because the package didn't provide source distribution on PyPi but this has been resolved

TheRealHaoLiu reviewed (commented):
```
### hiredis

The hiredis 2.1.0 release doesn't provide source distribution on PyPI which prevents users to build that python package from the
sources.
Downgrading to 2.0.0 (which provides source distribution) until the channels-redis issue is fixed or a newer hiredis version is
available on PyPi with source distribution.

* https://github.com/redis/hiredis-py/issues/138
```

this is still in the UPGRADE BLOCKERs doc

can u check on it

TheRealHaoLiu commented:
also... is this requirement needed? or can just be deleted if its a indirect dependency it should be ok to just delete it

jessicamack committed:
remove UPGRADE BLOCKER in README

jessicamack committed:
remove hiredis from requirements

we don't use hiredis directly so removing it from requirements. it will still be in requirements via aioredis

jessicamack commented:
@TheRealHaoLiu I think you're correct about it being an indirect dependency so I've removed it.

TheRealHaoLiu commented on requirements/requirements.txt:
```diff
@@ -201,10 +201,8 @@ grpcio==1.62.2
-    #   -r /awx_devel/requirements/requirements.in
-    #   aioredis
+hiredis==2.3.2
+    # via aioredis
```
since aioredis is removed from the channel_redis upgrade to 4.x we can probably just remove hiredis

TheRealHaoLiu replied:
updator need to be rerun after the channel_redis upgrade is merged

TheRealHaoLiu replied:
@jessicamack can u roll this into the channel_redis PR?

jessicamack replied:
yes, we'll have to merge them in order but we can leave this one for last and run update so the requirements.txt is up to date

jessicamack commented:
This work is being merged into #15329
